src/include/insn.h

```diff
@@ -102,11 +102,89 @@ NES_INSN_DEFN(asl) {
   }
 
   reg_block.update_z_flag(new_value);
-  if (new_value & 0b10000000) {
+  if (new_value & 0x80) {
     reg_block.update_n_flag(new_value);
   }
   reg_block.update_c_flag(old_value & 0x80);
   return true;
 }
 
+
+NES_INSN_DEFN(and) {
+  Reg8 old_value;
+  Reg8 new_value;
+  switch (opcode) {
+  case 0x29: { // Immediate
+    Data16 addr = mem.read_rom(reg_block.m_pc + 1, 1);
+    old_value = reg_block.m_accm;
+    new_value = old_value & addr;
+    reg_block.m_accm = new_value;
+    break;
+  }
+  case 0x25: { // Zero Page
+    Data16 addr = mem.read_rom(reg_block.m_pc + 1, 1);
+    old_value = mem.read(addr, 1);
+    new_value = reg_block.m_accm & old_value;
+    reg_block.m_accm = new_value;
+    break;
+  }
+  case 0x35: { // Zero Page, X
+    Data16 addr = mem.read_rom(reg_block.m_pc + 1, 1);
+    Reg8 xreg = reg_block.m_index_x;
+    old_value = mem.read(addr + xreg, 1);
+    new_value = old_value & reg_block.m_accm;
+    reg_block.m_accm = new_value;
+    break;
+  }
+  case 0x2D: { // Absolute
+    old_value = mem.read(addr, 1);
+    new_value = old_value & reg_block.m_accm;
+    reg_block.m_accm = new_value; 
+    break;
+  }
+  case 0x3D: { // Absolute, X
+    Data16 addr = mem.read_rom(reg_block.m_pc + 1, 2);
```

Comment:
Absolute has a 2 Byte operand which are actually swapped. So we need to swap them around with the utility function we have.